File: src/3046.split-the-array.3324/solution.ts
function isPossibleToSplit(nums: number[]): boolean {
    const set1 = new Set<number>();
    const set2 = new Set<number>();
    for (const n of nums) {
        if (set2.has(n)) {
            return false;
        }
        if (set1.has(n)) {
            set2.add(n)
        } else {
            set1.add(n)
        }
    }
    return true;
};

File: src/3046.split-the-array.3324/solution.java
class Solution {
    public boolean isPossibleToSplit(int[] nums) {
        int[] counts = new int[101];
        for(int n : nums){
            counts[n]++;
            if(counts[n] == 3){
                return false;
            }
        }
        return true;
    }
}